kgforge/specializations/models/rdf/service.py

```diff
@@ -128,18 +143,34 @@ def lookup_shape_from_node(self, node: URIRef) -> Shape:
 
 class RdfService:
 
-    def __init__(self, graph: Graph, context_iri: Optional[str] = None) -> None:
+    def __init__(self, graph: RDFDataset, context_iri: Optional[str] = None) -> None:
 
         if context_iri is None:
             raise ConfigurationError("RdfModel requires a context")
         self._graph = graph
+        self._sg = None
+        self._init_shape_graph_wrapper()
+        self.NXV = Namespace("https://bluebrain.github.io/nexus/vocabulary/")
         self._context_cache = {}
-        self.classes_to_shapes = self._build_shapes_map()
         resolved_context = self.resolve_context(context_iri)
         self.context = Context(resolved_context, context_iri)
-        self.types_to_shapes: Dict = self._build_types_to_shapes()
+        (
+            self.class_to_shape,
```

Comment:
I think for all 3 of those attributes there should be "getters" (for key access) that support both str and URIRef

Reply:
Reevaluate this. I think at service level adding getters would bring overhead.